lib/thrift/binary/framed/client.ex

```diff
@@ -252,8 +211,8 @@ defmodule Thrift.Binary.Framed.Client do
       {:error, :timeout} = timeout ->
         {:reply, timeout, s}
 
-      {:error, :closed} = err ->
-        {:disconnect, {:retry, err}, %{s |last_message: {call_args, caller}}}
+      {:error, :closed} = error ->
+        {:disconnect, error, s}
```

Comment:
We should rely on next clause to handle this, as we need to reply as well and would be identical behavior.

Reply:
Can we also do the same for : timeout above? The logic doesn't seem right there...

Reply:
I had generalized it by some tests failed. They seemed sensitive to the 3- vs. 4- element version of `:disconnect`.

I didn’t look further for this particular change but agree there’s still room for improvement.

Reply:
I just took another pass at this, and there's something about the `{:error, :closed}` case that appears to be different than the other errors. The current tests only work if we defer the reply until `disconnect/2` sends its `:stop`; if we reply immediately, something goes amiss and the process don't exit.

Reply:
Do you mean that the call does not get an exit here: https://github.com/pinterest/elixir-thrift/pull/303/files#diff-095d95cc2c507ddb3650fc4f4887facaR355? I think that it is fine for the call not to get an exit there, and instead wait for the `:EXIT` from the client:
```elixir
assert Client.friend_ids_of(client, 1234) == {:error, :closed}
assert_receive {:EXIT, ^client, {:error, :closed}}
```